src/Services/DocumentTemplates/DocumentTemplateRender.php

```diff
@@ -109,7 +109,15 @@ public function doRender($template_id, $template_content = null, $json_data = nu
         // purify html (and remove js)
         $isLegacy = stripos($template, 'portal_version') === false;
         $config = HTMLPurifier_Config::createDefault();
-        $config->set('Cache.SerializerPath', $GLOBALS['temporary_files_dir']);
+        $purifyTempFile = $GLOBALS['temporary_files_dir'] . DIRECTORY_SEPARATOR . 'htmlpurifier';
+        if (
+            !file_exists($purifyTempFile) &&
+            !is_dir($purifyTempFile)
+        ) {
+            mkdir($purifyTempFile);
```

Comment:
not sure it matters, but this will fail if $purifyTempFile is a regular file, you could grab the error and log it, but again pretty trivial thing.